<?xml version="1.0"?>
<launch>
    <!-- main_loop logs. Can be 'log' to be stored in log file or 'screen' to be printed to screen. -->
    <arg name="main_loop_output" default="screen" />
    <arg name="wandb" default="false" />

    <!-- Sets the obstacle representation type, accepted types are: "ellipse", "wedge","circles", "hybrid_circle", and "hybrid_ellipse" (full list in getObstacles())-->
    <arg name="obstacle_type" default="hybrid_circle" />
    <param name="obstacle_type" type="string" value="$(arg obstacle_type)" />

    <!-- Set if screenshots should be taken or not -->
    <arg name="screenshot" default="false" />
    <param name="screenshot" type="bool" value="$(arg screenshot)" />

    <!-- Set if pathfinding problem should be plotted or not -->
    <arg name="plot_pathfinding_problem" default="false" />
    <param name="plot_pathfinding_problem" type="bool" value="$(arg plot_pathfinding_problem)" />

    <!-- Sets planner type, accepted types include: "RRTStar", "RRTSharp","RRTXStatic" (full list in allocatePlanner()) -->
    <arg name="planner_type" default="RRTStar" />
    <param name="planner_type" type="string" value="$(arg planner_type)" />

    <!-- Setup mocks. View launch_all_mocks.launch for argument descriptions. -->
    <arg name="initial_speedup" default="1.0" />
    <arg name="num_ais_ships" default="5" />
    <arg name="ais_file" default="" />
    <arg name="gps_file" default="" />
    <arg name="goal_file" default="" />
    <arg name="wind_file" default="" />
    <arg name="random_seed" default="" />
    <arg name="smooth_changes" default="true" />
    <arg name="ocean_current_speed" default="0.0" />
    <arg name="ocean_current_direction" default="0.0" />
    <arg name="enable_los" default="false" />
    <arg name="globalWindSpeedKmph" default="1.0" />
    <arg name="globalWindDirectionDegrees" default="45.0" />
    <param name="enable_los" type="bool" value="$(arg enable_los)" />
    <include file="$(find local_pathfinding)/launch/launch_all_mocks.launch">
      <arg name="initial_speedup" value="$(arg initial_speedup)"/>
      <arg name="num_ais_ships" value="$(arg num_ais_ships)"/>
      <arg name="ais_file" value="$(arg ais_file)"/>
      <arg name="gps_file" value="$(arg gps_file)"/>
      <arg name="goal_file" value="$(arg goal_file)"/>
      <arg name="wind_file" value="$(arg wind_file)"/>
      <arg name="random_seed" value="$(arg random_seed)" />
      <arg name="smooth_changes" value="$(arg smooth_changes)" />
      <arg name="ocean_current_speed" default="$(arg ocean_current_speed)" />
      <arg name="ocean_current_direction" default="$(arg ocean_current_direction)" />
      <arg name="globalWindSpeedKmph" default="$(arg globalWindSpeedKmph)" />
      <arg name="globalWindDirectionDegrees" default="$(arg globalWindDirectionDegrees)" />
    </include>

    <node pkg="local_pathfinding" type="main_loop.py" name="main_loop" output="$(arg main_loop_output)"/>
    <node pkg="local_pathfinding" type="global_wind.py" name="global_wind" output="screen"/>
    <node pkg="local_pathfinding" type="local_path_visualizer.py" name="local_path_visualizer" />
    <node pkg="local_pathfinding" type="wandb_log.py" name="wandb_log" if="$(arg wandb)" output="screen" />
</launch>

